cpg-language-llvm/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/llvm/DeclarationHandler.kt

```diff
@@ -219,15 +217,17 @@ class DeclarationHandler(lang: LLVMIRLanguageFrontend) :
             // there are no names, so we need to invent some dummy ones for easier reading
             val fieldName = "field_$i"
 
+            frontend.scopeManager.enterScope(record)
+
             val field = newFieldDeclaration(fieldName, fieldType, listOf(), null, false)
 
             frontend.scopeManager.addDeclaration(field)
-        }
 
-        frontend.scopeManager.leaveScope(record)
+            frontend.scopeManager.leaveScope(record)
```

Comment:
Why do we enter and leave the record's scope for each of the fields separately now?

Reply:
Because of the way parsing of "classes" work in LLVM-IR. We handle the "type" of each field and if the type is a structure, we create a new class. If we would just enter the record once, we would create the class-type of the field within the scope of the fields' record instead of the global scope. I am not sure why this worked before, but if I leave it like it was the name of the class was wrong and it was created in the wrong scope.